src/lualib/ObjectAssign.ts

```diff
@@ -1,15 +1,12 @@
 // https://tc39.github.io/ecma262/#sec-object.assign
 // eslint-disable-next-line @typescript-eslint/ban-types
-export function __TS__ObjectAssign<T extends object>(this: void, to: T, ...sources: object[]): T {
-    if (to === undefined) {
```

Comment:
Can we just get rid of this check? Wouldn't this cause unexpected errors if `strictNullChecks` is disabled?